src/map/tileset.cpp

```diff
@@ -699,20 +720,66 @@ uint32_t CTileset::getQuadFromTile(graphic_index tile) const
 	return base | (base << 8) | (base << 16) | (base << 24);
 }
 
-void CTileset::fillSolidTiles(std::vector<unsigned int> *solidTiles) const
+std::vector<tile_index> CTileset::getAllTiles() const
 {
-	std::vector<int> seen_types;
-	seen_types.resize(solidTerrainTypes.size(), 0);
-	const size_t tilesCount = getTileCount();
-	for (size_t i = 16; i < tilesCount; i++) {
-		const CTileInfo &info = tiles[i].tileinfo;
-		if (info.BaseTerrain && info.MixTerrain == 0) {
-			if (seen_types[info.BaseTerrain] == 0) {
-				seen_types[info.BaseTerrain] = 1;
-				solidTiles->push_back(tiles[i].tile);
+	std::vector<tile_index> result;
+
+	tile_index tileIdx = 0;
+	for (auto &tile : tiles) {
+		if (tileIdx >= 0x10) { /// First 16 tiles for fog of war.
+			if (tile.tile != 0) { /// Ccheck for separator between tiles
+				result.push_back(static_cast<tile_index>(tileIdx));
+			}
+		}
+		tileIdx++;
+	}
+	return result;
+}
+
+std::vector<tile_index> CTileset::getSolidTiles() const
+{
+	std::vector<tile_index> result;
+
+	std::set<terrain_typeIdx> addedTerrains;
+	tile_index tileIdx = 0;
+	for (const auto &tile : tiles) {
+		if (tileIdx >= 0x10) { // First 16 tiles for fog of war
+			const CTileInfo &info = tile.tileinfo;
+			if (info.BaseTerrain && info.MixTerrain == 0
+				&& addedTerrains.count(info.BaseTerrain) == 0) {
+
+				addedTerrains.insert(info.BaseTerrain);
+				result.push_back(static_cast<tile_index>(tileIdx));
+			}
+		}
+		tileIdx++;
+	}
+	return result;
+}
+
+std::vector<tile_index> CTileset::getFirstOfItsKindTiles() const
+{
+	std::vector<tile_index> result;
+
+	bool foundFirstOfAKind = false;
+	tile_index tileIdx = 0;
+	for (auto &tile : tiles) {
+		if (tileIdx > 0xF) { // First 16 tiles for fog of war
```

Comment:
Be consistent with above methods, either `>= 0x10` or `> 0xF` even if result is identical

Reply:
fixed:
https://github.com/Wargus/stratagus/pull/732/commits/8bca15d592ce63e646cd6a7c829bad657471835c